File: jscomp/test/polyvar_match_test.ml
type 'a u = A of 'a option 
let f x = 
  match x with 
  | A (Some `a) -> 0 
  | A None -> 1
  | A (Some `c) -> 2 
  | A (Some (`d _)) -> 3 
  | A (Some (`e _)) -> 4  

File: jscomp/test/polyvar_match_test.js
'use strict';


function f(x) {
  var match = x._0;
  if (match !== undefined) {
    if (typeof match === "number") {
      if (match === /* c */99) {
        return 2;
      } else {
        return 0;
      }
    } else if (match.HASH === /* e */101) {
      return 4;
    } else {
      return 3;
    }
  } else {
    return 1;
  }
}

exports.f = f;
/* No side effect */
